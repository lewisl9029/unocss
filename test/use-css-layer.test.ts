import { createGenerator } from '@unocss/core'
import { describe, expect, it } from 'vitest'

describe('use-css-layer', () => {
  it('static', async () => {
    const uno = createGenerator({
      rules: [
        ['a', { name: 'bar1' }, { layer: 'a' }],
        ['b', { name: 'bar2' }, { layer: 'b' }],
        [/^c(\d+)$/, ([, d]) => ({ name: d }), { layer: 'c' }],
        [/^d(\d+)$/, ([, d]) => `/* RAW ${d} */`, { layer: 'd' }],
      ],
      shortcuts: [
        ['abcd', 'a b c2 d3', { layer: 's' }],
        ['abc', 'a b c4'],
      ],
      presets: [],
      outputToCssLayers: true,
    })
    const { css } = await uno.generate('a b abc abcd d4 c5', { preflights: false })
    expect(css).toMatchSnapshot()
  })
})
