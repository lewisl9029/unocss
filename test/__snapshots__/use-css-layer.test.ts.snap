// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`use-css-layer > static 1`] = `
"/* layer: shortcuts */
@layer shortcuts{
.abc{name:bar1;name:bar2;name:4;}
}
/* layer: a */
@layer a{
.a{name:bar1;}
}
/* layer: b */
@layer b{
.b{name:bar2;}
}
/* layer: c */
@layer c{
.c5{name:5;}
}
/* layer: d */
@layer d{
/* RAW 4 */
/* RAW 3 */
}
/* layer: s */
@layer s{
.abcd{name:bar1;name:bar2;name:2;}
}"
`;
